### ISSUE TITLE
System.Drawing.Common is not supported on non-Windows platforms

### ISSUE THREAD

**Issue by qcjxberin** [dotnet/runtime#80148]

**Comment by danmoseley** [comment]
@siddjain do you have to use System.Drawing? As mentioned, it is out of support now.

**Comment by siddjain** [comment]
Thanks all for the help. re: do you have to use System.Drawing? to answer literally, its a loaded question. One can always port the code to another library. In fact, I already did it for one of my GDI+ apps as I wanted to learn Swift but to my surprise the C# code runs much faster than Swift!

**Comment by danmoseley** [comment]
> Also what I understand when someone says the library is out of support is that no one is fixing bugs and issues but anyone is welcome to use it as-is and it will continue to be available and downloadable from the internet. is that correct?

System.Drawing is not going anywhere, but the code inside it that uses libgdiplus has been removed, which means that it will only work on Windows now. 

* in .NET 6, you can enable mac/Linux support, but it is off by default. see See https://learn.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/6.0/system-drawing-common-windows-only
* in .NET 7 and later, it does not exist.

 .NET 6 will go out of support in due course, although you'll still be able to download it if you don't mind being out of support.

In short, I recommend you don't write code against System.Drawing unless it's only intended to run on Windows. The only exception might be if it's just a temporary thing, in which case you can use .NET 6. But generally you should pick another library to do this stuff unless you're Windows only.

**Comment by siddjain** [comment]
Thanks Dan for the clarification!

**Comment by janseris** [comment]
@siddjain 
>can always port the code to another library
This part is not magically for free you know

**Comment by siddjain** [comment]
Correct that’s is why it’s a loaded question.

Sent from my iPhone

On Jan 4, 2023, at 11:48 AM, janseris ***@***.***> wrote:

﻿

@siddjain<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fsiddjain&data=05%7C01%7C%7Cbf900490cff64091c4c608daee8cb75e%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638084585393937304%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=crtyb493yiBDO7zBD8O6aIV5%2Fw9BqF80I%2BqONXduMTI%3D&reserved=0>

one can always port the code to another library

This part is not magically for free you know

**Comment by Rombersoft** [comment]
Reading it all I came to the conclusion that I learned not the right programming language. It was necessary to learn Java 10 years ago when C# finished with such valuable problems. Microsoft is a great company but it is not able to make support graphics basics for three OS. I will keep silent about cross-platformed desktop GUI, they are missing. Miguel De Icaza come back. If you advise migrating to MAUI or Skia, maybe will you write a wrapper for it inside Linux System.Drawing implementation?
